src/server/api/consume-wifi-data.ts

```diff
@@ -75,12 +77,12 @@ async function consumeLastBatch({ client }: { client: SupabaseClient }) {
         batch.messages
           .filter((message) => message.value)
           .map((message) =>
-            registry
-              .decode(message.value)
-              .then((decodedValue) => parseMessage({
+            registry.decode(message.value!).then((decodedValue) =>
+              parseMessage({
                 timestamp: message.timestamp,
                 decodedValue,
-              }))
+              })
+            )
```

Comment:
Think this is also harder to read, prefer the indented chain of functions. I do see I missed a bracket around the map :see_no_evil: